### PR TITLE
Add extra metrics and consul metadata

### PR DESCRIPTION
**Pull request by bmerry**
This is to support a more fully-featured dashboard.

### CODE REVIEW

**Comment by bmerry** [comment]
It's just occurred to me that the dsims might also benefit from the same treatment, although there it is less necessary since for dsims the antenna name gets baked into the task name.

File: src/katsdpcontroller/generator.py [hunk 1]
```diff
@@ -798,6 +798,9 @@ def _make_fgpu(
         fgpu.pass_telstate = False
         fgpu.data_suspect_sensor = data_suspect_sensor
         fgpu.data_suspect_range = (2 * i, 2 * i + 2)
+        # Identify the antenna, if the input labels are consistent
+        if input_labels[0][:-1] == input_labels[1][:-1]:
```
Comment (james-smith-za) [inline src/katsdpcontroller/generator.py:802]: Is this just a sanity-check to see whether both pols are coming from the same antenna?

File: src/katsdpcontroller/generator.py [hunk 2]
```diff
@@ -831,10 +834,11 @@ def _make_fgpu(
                 "rx-missing-unixtime",
             ]:
                 fgpu.sensor_renames[f"input{j}-{name}"] = f"{stream.name}-{label}-{name}"
-        # Prepare expected data rate
+        # Prepare expected data rates etc
         fgpu.static_gauges["fgpu_expected_input_heaps_per_second"] = sum(
             src.adc_sample_rate / src.samples_per_heap for src in srcs
         )
+        fgpu.static_gauges["fgpu_expected_engines"] = 1.0
```
Comment (james-smith-za) [inline src/katsdpcontroller/generator.py:841]: I'm ... not sure why just 1? Ditto for the xbgpu case down below.

File: src/katsdpcontroller/generator.py (hunk 1 above)
Comment (bmerry) [reply src/katsdpcontroller/generator.py:802]: Yes. If an engine is being fed by different antennas (or someone is using it with sources that don't follow the MK input naming convention), avoid producing misleading antenna information.

File: src/katsdpcontroller/generator.py (hunk 2 above)
Comment (bmerry) [reply src/katsdpcontroller/generator.py:841]: `fgpu` is a single engine. The number of expected engines per engine is 1. The code that processes static_gauges takes a sum across all the engines.

Comment (james-smith-za) [reply src/katsdpcontroller/generator.py:841]: Ah. That explains things.

**Review by james-smith-za — APPROVED** [review]

**Pull request merged**